qcodes_contrib_drivers/drivers/Bilt/ITest.py

```diff
@@ -0,0 +1,475 @@
+
+from typing import Union, Tuple, Any
+from functools import partial
+from math import ceil
+from time import sleep
+
+from qcodes.instrument.base import Instrument
+from qcodes.instrument.channel import InstrumentChannel, ChannelList
+from qcodes.instrument.channel import MultiChannelInstrumentParameter
+from qcodes.instrument.visa import VisaInstrument
+from qcodes.utils import validators as vals
+
+
+class iTestChannel(InstrumentChannel):
+    """
+    A single channel of iTest.
+    """
+    
+    def __init__(self, parent: Instrument,
+                       name: str,
+                       chan_num: int) -> None:
+        """
+        Args:
+            parent: The instrument to which the channel is attached.
+            name: The name of the channel.
+            chan_num: The number of the channel in question
+        """
+        super().__init__(parent, name)
+        
+        self.chan_num = chan_num
+        # Get channel id
+        i, c = round(chan_num/4.)+1, chan_num%4
+        self.chan_id = 'i{};c{};'.format(i,c)
+        
+        self.add_parameter('v',
+                           label='Channel {} voltage'.format(chan_num),
+                           unit='V',
+                           docstring='Voltage of the channel in volt.',
+                           get_cmd=partial(self._parent._get_voltage, chan_num),
+                           get_parser=float,
+                           set_cmd=partial(self._parent._set_voltage, chan_num),
+                           vals=vals.Numbers(-50, 50)
+                           )
+           
+        self.add_parameter('i',
+                           label='Channel {} current'.format(chan_num),
+                           unit='A',
+                           docstring='Current of the channel in ampere.',
+                           get_cmd=partial(self._parent._get_current, chan_num),
+                           get_parser=float
+                           )
+        
+        self.add_parameter('ramp_slope',
+                           label='Channel {} ramp slope'.format(chan_num),
+                           unit='V/ms',
+                           docstring='Slope of the ramp in V/ms.',
+                           get_cmd=partial(self._parent._get_ramp_slope, chan_num),
+                           get_parser=float,
+                           set_cmd=partial(self._parent._set_ramp_slope, chan_num),
+                           vals=vals.Numbers(0, 1)
+                           )
+        
+        self.add_parameter('output_mode',
+                           label='Channel {} output mode'.format(chan_num),
+                           docstring='Mode of the output {exp, ramp}.',
+                           get_cmd=partial(self._parent._get_output_function, chan_num),
+                           get_parser=str,
+                           set_cmd=partial(self._parent._set_output_function, chan_num),
+                           set_parser=str
```

Comment:
is it true that the output_mode can only have a certain strings as values? for example, only "ramp" and "non-ramp". If so, then it will be very useful for users and for correctness to use an Enum validator with those strings:
```suggestion
                           set_parser=str,
                           vals=vals.Enum("ramp", "non-ramp")
```

Reply:
This is recommended.